emscripten.py

```diff
@@ -358,8 +344,13 @@ def emscript(infile, outfile_js, memfile, temp_files, DEBUG):
     pre += '\n' + global_initializers + '\n'
 
   if shared.Settings.RELOCATABLE:
+    static_bump = align_memory(webassembly.parse_dylink_section(infile)[0])
+    memory = Memory(static_bump)
```

Comment:
I'm pleasantly surprised that we only need the Memory stuff in relocatable mode now...

Reply:
I know its awesome isn't it!

In theory we could even parse the wasm at runtime and avoid JS ever knowing this information a-priori.

These kind of changes really help make the JS less bound to given wasm file (@surma @RReverser will be happy!).